protocol/x/assets/keeper/asset.go

```diff
@@ -33,9 +33,23 @@ func (k Keeper) CreateAsset(
 		)
 	}
 
-	// Ensure assetId zero is always USDC. This is a protocol-wide invariant.
-	if assetId == types.AssetUsdc.Id && denom != types.AssetUsdc.Denom {
-		return types.Asset{}, types.ErrUsdcMustBeAssetZero
+	if assetId == types.AssetUsdc.Id {
+		// Ensure assetId zero is always USDC. This is a protocol-wide invariant.
+		if denom != types.AssetUsdc.Denom {
+			return types.Asset{}, types.ErrUsdcMustBeAssetZero
+		}
+
+		// Confirm that USDC asset has the expected denom exponent (-6).
+		// This is an important invariant before coin-to-quote-quantum conversion
+		// is correctly implemented (CLOB-871).
```

Comment:
nit: because at the first glance, I thought this was a TODO item
```suggestion
		// is correctly implemented. See CLOB-871 for details.
```